### PR TITLE
Remove backported revcheck compatibility

### PR DESCRIPTION
**Pull request by alfsb**
This PR removes some of backported revcheck behaviors, that seems incorrect or makes count debugging difficult. They *new* behaviors are:

1. Tracks `chmonly/` as translatable;
2. Tracks `extensions.xml` as translatable;
3. Counts and lists ok and old exclusively.

The first change will include 4 files as translatable, the second will mark one file as translatable and possibly a few as NotInEn. The last does not change file tracking or listing, but it causes a lot of changes on translator listing.

What it does is counting Ok and Old cases in separated counters, and push all other situations on a "Other" counter. So after this, file count and file listing of outdated cases will match, and RevtagProblem will not be counted as Outdated.

Note that RevtagProblem cannot be listed alongside with Outdated anyways, because it's not possible to generate diff links in this case. This changes will make count debug vastly easier, as it removes all instances of *possibly not* counted cases in a lot of corner cases.

Reviews and commentary welcome. I plan to push this PR next week if there are no objections.

This PR supersedes https://github.com/php/doc-base/issues/175.

### CODE REVIEW

File: scripts/translation/lib/RevcheckIgnore.php [hunk 1]
```diff
@@ -57,13 +57,6 @@ public static function ignore( $filename ) : bool
         if ( $filename == "reference/datetime/timezones.xml" )
             return true;
 
-        if ( IGNORE_EXTENSIONS_XML )
-            if ( str_ends_with( $filename , '/extensions.xml') ) // track/count backport
-                return true;
```
Comment (Girgias) [inline scripts/translation/lib/RevcheckIgnore.php:62]: Is it possible to hardcode this like the `reference/datetime/timezones.xml` file as:
```php
if ( $filename == "appendices/extensions.xml" )
            return true;
```
?

File: scripts/translation/lib/RevcheckIgnore.php [hunk 2]
```diff
@@ -57,13 +57,6 @@ public static function ignore( $filename ) : bool
...
-        if ( IGNORE_CHMONLY_DIR )
-            if ( str_starts_with( $filename , 'chmonly/') )      // track/count backport
-                return true;
```
Comment (Girgias) [inline scripts/translation/lib/RevcheckIgnore.php:65]: Similarly can we hardcode the 4 files?

File: scripts/translation/lib/RevcheckIgnore.php (hunk 1 above)
Comment (alfsb) [reply scripts/translation/lib/RevcheckIgnore.php:62]: I'm trying to make exceptions as few as possible. And in this case, I also think that the original ignore was as mistake:
```
$ find * | grep "/extensions.xml"
appendices/migration56/extensions.xml
appendices/extensions.xml
```
All files ending in "/extensions.xml" are translatable.

File: scripts/translation/lib/RevcheckIgnore.php (hunk 2 above)
Comment (alfsb) [reply scripts/translation/lib/RevcheckIgnore.php:65]: In the same theme of having as few exceptions as possible, this change makes `chmonly/` a normal translatable directory, as it seems to me (albeit the name). This directory only contains the four .xml files, all translatable.

It is no different from `chapters/` or ` faq/` in that matter. And with CHM manuais building again, the files in this directory are being included, untranslated, for all translations.

File: scripts/translation/lib/RevcheckIgnore.php (hunk 1 above)
Comment (Girgias) [reply scripts/translation/lib/RevcheckIgnore.php:62]: `appendices/extensions.xml` should NOT be translated, it is a generated file. Which is identical to `reference/datetime/timezones.xml` this is why I asked to hardcode the exception.

I agree that not marking `appendices/migration56/extensions.xml` as translatable is a bug.

File: scripts/translation/lib/RevcheckIgnore.php (hunk 2 above)
Comment (Girgias) [reply scripts/translation/lib/RevcheckIgnore.php:65]: I don't think CHM builds other than English are provided, thus I don't see the point in making those files available for translation. @jimwins do you know if CHM builds can work with translations.

Comment (jimwins) [reply scripts/translation/lib/RevcheckIgnore.php:65]: We do build CHM files for other languages (not Italian and Ukrainian for some reason). @mumumu does the builds, maybe they'd know if translated files in `chmonly` would be picked up. It seems to me that they should be, but I can't build the CHM files so I can't confirm.

File: scripts/translation/lib/RevcheckIgnore.php (hunk 1 above)
Comment (alfsb) [reply scripts/translation/lib/RevcheckIgnore.php:62]: Oh, I see the confusion. `appendices/extensions.xml` is already ignored (see line 53), and removing the blank ignore on `/extensions.xml` will not change this. So it's already done.

File: scripts/translation/lib/RevcheckIgnore.php (hunk 2 above)
Comment (alfsb) [reply scripts/translation/lib/RevcheckIgnore.php:65]: I downloaded the chm file for Brazilian Portuguese from https://www.php.net/download-docs.php , extracted it and grep'ed it for `The Windows HTML Help Edition of the PHP Manual`, with a match. So this files are being included on translated chm manual builds. The page indicates builds as recent as 8 Nov 2024, if I ready ir correctly.

Comment (alfsb) [reply scripts/translation/lib/RevcheckIgnore.php:65]: Update. I now confirmed all 4 files of `chmonly` are included and linked on internal CHM spine and HTML rendered ToC.

Comment (Girgias) [reply scripts/translation/lib/RevcheckIgnore.php:65]: ACK, then this is fine.

**Comment by alfsb** [comment]
Merged. Now, local revcheck will slight differ from hosted revcheck, as it is tracking `chmonly` and some `reserved.xml` files, and also counting changes on translator summary.

I do plan make two, possibly three changes (robust log parsing, strict rev-skip positioning, visual tweaks), beyond the already opened PR for sequential skip-revcheck.

Let me know if systems people want to wait to all changes before updating, or want to mentioned or have issues opened in each instance.

**Pull request merged**